src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCaConfigurationTest.java

```diff
@@ -186,31 +185,50 @@ private void subscribeToCertificateUpdates(GetCertificateRequest request)
         api.subscribeToCertificateUpdates(request);
     }
 
+    private Map<String, URI> givenCAStoredOnFilesystem(PrivateKey pk, X509Certificate... chain) throws IOException,
+            CertificateEncodingException, URISyntaxException {
+        Path pkPath = rootDir.resolve("private.key").toAbsolutePath();
+        CertificateTestHelpers.writeToPath(
+                pkPath, PEM_BOUNDARY_PRIVATE_KEY, Collections.singletonList(pk.getEncoded()));
+
+        List<byte[]> encodings = new ArrayList<>();
+        for (X509Certificate x509Certificate : chain) {
+            byte[] encoded = x509Certificate.getEncoded();
+            encodings.add(encoded);
+        }
+
+        Path chainPath = rootDir.resolve("certificate.pem").toAbsolutePath();
+        CertificateTestHelpers.writeToPath(
+                chainPath.toAbsolutePath(), PEM_BOUNDARY_CERTIFICATE, encodings);
+
+        return new HashMap<String, URI>(){{
+            put("privateKey", pkPath.toUri());
+            put("certificateAuthority", chainPath.toUri());
+        }};
```

Comment:
This HashMap initialization technique creates an extra anonymous class which holds hidden references to the enclosing object, and might cause memory leak issues. We should try to avoid it.